packages/extension-dapp/src/bundle.ts

```diff
@@ -61,8 +61,8 @@ function getWindowExtensions (originName: string): Promise<[InjectedExtensionInf
   );
 }
 
-// enables all the providers found on the injected window interface
-export function web3Enable (originName: string, compatInits: (() => Promise<boolean>)[] = []): Promise<InjectedExtension[]> {
+// enables all the providers found on the injected window interface, or just the named one
+export function web3Enable (originName: string, extensionName?: string, compatInits: (() => Promise<boolean>)[] = []): Promise<InjectedExtension[]> {
```

Comment:
```suggestion
export function web3Enable (originName: string, compatInits: (() => Promise<boolean>)[] = [], extensionName?: string): Promise<InjectedExtension[]> {
```

Should be added to the back since moving it around would break it for users that already use the compat (and don't use type checking)

Reply:
Just to confirm, that I can just pass an empty array and keep the same behaviour as not having it

Reply:
100%. Empty array means there is nothing to apply.